superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -42,7 +47,7 @@ export function datasetReducer(
     case DatasetActionType.selectSchema:
       return {
         ...trimmedState,
-        ...action.payload,
+        schema: action.payload as string,
```

Comment:
```suggestion
        [action.payload.name]: action.payload.value,
```
(2/3): By setting this with `DatasetReducerPayloadType` in types, it'll already be defined as a `string`